packages/evm/src/eof/container.ts

```diff
@@ -0,0 +1,461 @@
+  dataSize: number // Size of the data section
+  dataSizePtr: number // Used to edit the dataSize in RETURNCONTRACT
+  buffer: Uint8Array // The raw buffer of the entire header
+
+  private codeStartPos: number[] // Internal array to track at which byte of the container the code starts (per section)
+
+  /**
+   * Create an EOF header. Performs various validation checks inside the constructor
+   * @param input The input should either be a raw header, or a complete container
+   */
+  constructor(input: Uint8Array) {
+    if (input.length > MAX_HEADER_SIZE) {
+      throw new Error('err: container size more than maximum valid size')
+    }
+    const stream = new StreamReader(input)
+    // Verify that the header starts with 0xEF0001
+    stream.verifyUint(FORMAT, EOFError.FORMAT)
+    stream.verifyUint(MAGIC, EOFError.MAGIC)
+    stream.verifyUint(VERSION, EOFError.VERSION)
+    if (input.length < 15) {
+      throw new Error('err: container size less than minimum valid size')
+    }
+    // Verify that the types section is present, and verify that the type section length is valid
+    stream.verifyUint(KIND_TYPE, EOFError.KIND_TYPE)
+    const typeSize = stream.readUint16(EOFError.TypeSize)
+    if (typeSize < TYPE_MIN) {
+      validationError(EOFError.InvalidTypeSize, typeSize)
+    }
+    if (typeSize % TYPE_DIVISOR !== 0) {
+      validationError(EOFError.InvalidTypeSize, typeSize)
+    }
+    if (typeSize > TYPE_MAX) {
+      throw new Error(`err: number of code sections must not exceed 1024 (got ${typeSize})`)
+    }
+    // Verify that the code section is present, and verify that the code section size is valid
+    stream.verifyUint(KIND_CODE, EOFError.KIND_CODE)
+    const codeSize = stream.readUint16(EOFError.CodeSize)
+    if (codeSize < CODE_MIN) {
+      validationError(EOFError.MinCodeSections)
+    }
+    if (codeSize !== typeSize / TYPE_DIVISOR) {
+      validationError(EOFError.TypeSections, typeSize / TYPE_DIVISOR, codeSize)
+    }
+    // Read the actual code sizes in the code section, and verify that each code section has the minimum size
+    const codeSizes = []
+    for (let i = 0; i < codeSize; i++) {
+      const codeSectionSize = stream.readUint16(EOFError.CodeSection)
+      if (codeSectionSize < CODE_SIZE_MIN) {
+        validationError(EOFError.CodeSectionSize)
+      }
+      codeSizes.push(codeSectionSize)
+    }
+
+    // Check if there are container sections
+    let nextSection = stream.readUint()
+    const containerSizes: number[] = []
+    if (nextSection === KIND_CONTAINER) {
+      // The optional container section is present, validate that the size is within bounds
+      const containerSectionSize = stream.readUint16(EOFError.ContainerSize)
+
+      if (containerSectionSize < CONTAINER_MIN) {
+        validationError(EOFError.ContainerSectionSize)
+      }
+      if (containerSectionSize > CONTAINER_MAX) {
+        validationError(EOFError.ContainerSectionSize)
+      }
```

Comment:
As above, consolidate since error is the same
```suggestion
      if ((containerSectionSize < CONTAINER_MIN) || (containerSectionSize > CONTAINER_MAX)) {
        validationError(EOFError.ContainerSectionSize)
      }
```